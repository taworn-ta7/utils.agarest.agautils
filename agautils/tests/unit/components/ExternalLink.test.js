import testHelpers from '@@/unit/test-helpers.js';

import ExternalLink from '@/components/ExternalLink.vue';

describe('ExternalLink.vue', () => {
  const name = 'Test';
  const url = 'example.com';
  const fullUrl = 'https://' + url;

  const setupWrapper = async () => {
    const requiredProps = { name, url };
    const options = {
      props: requiredProps
    };
    const wrapper = await testHelpers.shallowMount(ExternalLink, options);
    return wrapper;
  };

  describe('Desktop', () => {
    test('Render default contents', async () => {
      const wrapper = await setupWrapper();

      expect(wrapper)
        .toMatchSnapshot();
    });

    test('Click event', async () => {
      const wrapper = await setupWrapper();

      await wrapper.find('a').trigger('click');

      expect(window.nw.Shell.openExternal)
        .toHaveBeenCalledWith(fullUrl);
    });
  });

  describe('Web', () => {
    beforeEach(() => {
      window.webSetup();
    });

    test('Render default contents', async () => {
      const wrapper = await setupWrapper();

      expect(wrapper)
        .toMatchSnapshot();
    });

    test('Click event', async () => {
      const wrapper = await setupWrapper();

      await wrapper.find('a').trigger('click');

      expect(window.open)
        .toHaveBeenCalledWith(fullUrl, '_blank');
    });
  });
});
