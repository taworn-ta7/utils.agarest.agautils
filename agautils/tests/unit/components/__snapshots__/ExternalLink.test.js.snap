// Vitest Snapshot v1

exports[`ExternalLink.vue > Desktop > Render default contents 1`] = `<a href="#" title="example.com">Test</a>`;

exports[`ExternalLink.vue > Web > Render default contents 1`] = `<a href="#" title="example.com">Test</a>`;
